src/idl_gen_cpp.cpp

```diff
@@ -2941,10 +2943,18 @@ class CppGenerator : public BaseGenerator {
           }
           case BASE_TYPE_UTYPE: {
             value = StripUnionType(value);
-            code += "_fbb.CreateVector<uint8_t>(" + value +
-                    ".size(), [](size_t i, _VectorArgs *__va) { "
-                    "return static_cast<uint8_t>(__va->_" +
-                    value + "[i].type); }, &_va)";
+            if (opts_.scoped_enums) {
+              auto type = Name(*field.value.type.enum_def);
+              code += "_fbb.CreateVector<" + type + ">(" + value +
+                      ".size(), [](size_t i, _VectorArgs *__va) { "
+                      "return __va->_" +
+                      value + "[i].type; }, &_va)";
+            } else {
+              code += "_fbb.CreateVector<uint8_t>(" + value +
```

Comment:
Is it possible to split this sentence into two parts: a prefix and a fixed main expression?
For example:
```cpp
const auto& type = opts_.scoped_enums ? Name(*field.value.type.enum_def) : std::string("uint8_t");
```

Reply:
Sure. Updated.
Though, as `static_cast` is another diff between them, latter part is not fully fixed expr.

Reply:
I'm sorry, I missed this `static_cast`.
Now code looks over-complicated (`cast_if_needed` lambda). Can you revert this commit or replace `cast_if_needed` by string value?

Reply:
OK, let me replace the lambda with a string.